### PR TITLE
[react-native-auth0] add types for var `user` returned by `useAuth0`

### PR DESCRIPTION
**Pull request by januswel**
Type definitions are based on OpenID Connect scope claims

https://openid.net/specs/openid-connect-core-1_0.html#ScopeClaims

And rewrite the types for function property of `useAuth0` return obj with arrow function style to fix errors from the ESLint rule @typescript-eslint/unbound-method

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/66020

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## react-native-auth0 ([<kbd>unpkg</kbd>](https://unpkg.com/browse/react-native-auth0@latest/))
was missing the following properties:
1. TimeoutError
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7cf71f7b558977fada6718e2b821c641cb71ce2d
</p>

**Comment by januswel** [comment]
Hi owners, please review my codes!!!

File: types/react-native-auth0/index.d.ts [hunk 1]
```diff
@@ -323,24 +324,62 @@ export default class Auth0 {
...
+    getCredentials: (scope?: string, minTtl?: number, parameters?: any) => Promise<Credentials>;
+    clearCredentials: () => Promise<void>;
+    requireLocalAuthentication: (
         title?: string,
         description?: string,
         cancelTitle?: string,
         fallbackTitle?: string,
         strategy?: LocalAuthenticationStrategy,
-    ): Promise<void>;
+    ) => Promise<void>;
```
Comment (gfriedm4) [inline types/react-native-auth0/index.d.ts:379]: I think the arrow functions here imply that these are bound functions, but I don't think they are?

`authorize`: https://github.com/auth0/react-native-auth0/blob/master/src/webauth/index.js#L70
`clearSession`: https://github.com/auth0/react-native-auth0/blob/master/src/webauth/index.js#L164
, etc.

Comment (januswel) [reply types/react-native-auth0/index.d.ts:379]: Thank you for your review ❤️ 

These are type definitions for `useAuth0`
The `authorize` function is defined in: https://github.com/auth0/react-native-auth0/blob/master/src/hooks/auth0-provider.js#L66

**Review by gfriedm4 — APPROVED** [review]
LGTM 👍

**Comment by januswel** [comment]
Ready to merge

**Pull request merged**